### DISCUSSION TITLE
Using ManagedActions in an x64 app

### DISCUSSION THREAD

**Discussion by NJullienSweet** [oleg-shilo/wixsharp#1647] [Q&A] [answered]
I need to check something with you.
From what I understood here https://github.com/oleg-shilo/wixsharp/issues/1427, you can't have an MSI and a Wix Bundle built with `Platform = Platform.x64` when using ManagedActions, because these are invoked in an x86 environment. I tried and ended up with a `BadImageFormatException` as well when invoking them.
So does that mean I need to give up on all the [useful automations you made in WixSharp with the platform](https://github.com/oleg-shilo/wixsharp/wiki/Deployment-scenarios#x64-components) (e.g. auto selection of Program Files) and install everything using the correct architecture?
Is that right or did I miss something to avoid removing the Platform.x64?

**Comment by oleg-shilo** [comment]
Hi Nicolas,

The situation is not as bad as it may seem.

Indeed, an MSI session that every setup is running on is an x86 process (MSI service). Meaning that every ManagedAction needs to be compiled with the target architecture either x86 or AnyCPU. 

Fortunately, all WixSharp assemblies and the WixSharp project templates do target compatible runtime already so we are good.

Your problem is most likely caused by your custom action (ManagedAction) assembly either compiled for x64 or it has Interop dependency on some x64 native DLL.

I just checked the bootstrapper and the MSIx64 from the samples and it works fine. So, check your assembly build.

**Reply by NJullienSweet** [reply, marked as answer]
Following your answer, I checked the csproj and tried commenting lines one by one. I was surprised to discover that the culprit was `<PlatformTarget>x86</PlatformTarget>`. When removed, the custom actions execute and the uninstallation where they are invoked completes. I still don't quite understand why this was messing with it. Anyway I can now continue, thanks to your pointers.
Thank you for your help, Oleg!
